.github/workflows/cli.test.yaml

```diff
@@ -43,18 +43,36 @@ jobs:
         with:
           node-version: "20.10.0"
 
-      - name: Install dependencies
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.21.5"
+          check-latest: true
+          cache-dependency-path: |
+            ./go-api/go.sum
+
+      - name: Install golang-migrate
+        run: |
+          curl -L https://github.com/golang-migrate/migrate/releases/download/v4.17.0/migrate.linux-amd64.tar.gz | tar xvz
+          sudo mv ./migrate /usr/bin
+
+      - name: Install yarn dependencies
```

Comment:
Can we move this closer to location where it is needed? Currently, it is in between go related steps.